cg/meta/archive/archive.py

```diff
@@ -81,17 +64,19 @@ def archive_spring_files_and_add_archives_to_housekeeper(
     ) -> None:
         """Archives all non archived spring files. If a limit is provided, the amount of files archived are limited
         to that amount."""
-
-        files_to_archive: list[File] = self.get_non_archived_and_non_pdc_spring_files()
-        files_and_samples: list[FileAndSample] = self.add_samples_to_files(files_to_archive)[
-            :spring_file_count_limit
-        ]
-
+        nr_of_files_handled: int = 0
         for archive_location in ArchiveLocations:
-            files_and_samples_for_location: list[FileAndSample] = filter_files_on_archive_location(
-                files_and_samples=files_and_samples, archive_location=archive_location
+            if spring_file_count_limit and nr_of_files_handled > spring_file_count_limit:
+                LOG.info(f"{spring_file_count_limit} files sent to archive - exiting.")
+            files_to_archive: list[File] = self.housekeeper_api.get_non_archived_spring_files(
+                tags=[archive_location],
+                limit=spring_file_count_limit - nr_of_files_handled
+                if spring_file_count_limit
+                else None,
```

Comment:
Not sure what to do here but it indeed reads quite difficult to me.

Reply:
Went with removing the nr_of_files_handled variable altogether